ui/apps/platform/src/Containers/SystemHealth/Components/DeclarativeConfigurationHeathCard.tsx

```diff
@@ -0,0 +1,117 @@
+import pluralize from 'pluralize';
+
+import IconText from 'Components/PatternFly/IconText/IconText';
+import { fetchDeclarativeConfigurationsHealth } from 'services/IntegrationHealthService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+import { IntegrationHealthItem } from '../utils/integrations';
+
+type Props = {
+    pollingCount: number;
+};
+
+function DeclarativeConfigurationHealthCard({ pollingCount }: Props): ReactElement {
+    const [isFetching, setIsFetching] = useState(false);
+    const [requestErrorMessage, setRequestErrorMessage] = useState('');
+    const [items, setItems] = useState<IntegrationHealthItem[]>([]);
+
+    useEffect(() => {
+        setIsFetching(true);
+        fetchDeclarativeConfigurationsHealth()
+            .then((itemsFetched) => {
+                setRequestErrorMessage('');
+                setItems(itemsFetched);
+            })
+            .catch((error) => {
+                setRequestErrorMessage(getAxiosErrorMessage(error));
+                setItems([]);
+            })
+            .finally(() => {
+                setIsFetching(false);
+            });
+    }, [pollingCount]);
+
+    /*
+     * Wait for isFetching only until response to the initial request.
+     * Otherwise count temporarily disappears during each subsequent request.
+     */
+    const hasCount = (pollingCount !== 0 || !isFetching) && !requestErrorMessage;
+    const itemCount = items.length;
+
+    return (
+        <Card isFullHeight isCompact>
+            <CardHeader>
+                <CardHeaderMain>
+                    <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                        <FlexItem>
+                            <CardTitle component="h2">Declarative configuration</CardTitle>
+                        </FlexItem>
+                        {hasCount && (
+                            <FlexItem>
+                                <IconText
+                                    icon={
+                                        itemCount === 0 ? (
+                                            <CheckCircleIcon color="var(--pf-global--success-color--100)" />
+                                        ) : (
+                                            <ExclamationCircleIcon color="var(--pf-global--danger-color--100)" />
+                                        )
+                                    }
+                                    text={
+                                        itemCount === 0
+                                            ? 'no errors'
+                                            : `${itemCount} ${pluralize('error', itemCount)}`
+                                    }
+                                />
+                            </FlexItem>
+                        )}
+                    </Flex>
+                </CardHeaderMain>
+            </CardHeader>
+            {(requestErrorMessage || itemCount !== 0) && (
+                <CardBody>
```

Comment:
small suggestion: Show CardBody with spinner when fetching

Reply:
Aha, that was my first draft, following our usual pattern. Here is are some issues that I found:

1. By design, card has body only if either declarative configuration errors or request error.
2. A possible design pivot so System Health page can display overview without scrolling (as cards continue to increase) is to follow example of Cluster health card displays counts only and has **View** link for remediation to move error table from card bodies to separate pages.

    That is if there are errors, then card headers have **View errors** links at right to pages that render respective error tables:

    * /main/system-health/declarative-configuration
    * /main/system-health/image-integrations
    * /main/system-health/notifier-integrations
    * /main/system-health/backup-integrations

    Having said that, one card for Integration Health instead of 3 cards seems like it scales better and might somewhat reduce the gap in information architecture between integration and cluster health

4. When I started to add conditional rendering for the spinner to the card header:
    * I noticed an annoying change every 30 seconds whether it rendered either instead of or in addition to the previous count.
    * Concerning the preceding comment, it seems like clearing an error message optimistically might cause similar periodic on-off cycle for a persistent problem.

I will let this simmer in mental slow cooker over the weekend.